### PR TITLE
Bug 2001817:  Failed to load RoleBindings list that will lead to ‘Role name’ is not able to be selected on Create RoleBinding page as well

### PR DESCRIPTION
**Pull request by cyril-ui-developer**
/assign @spadgett 
cc: @rawagner 

### Before:
 RoleBinding
Screen Shot 2021-09-13 at 5 45 45 PM

### After:
Screen Shot 2021-09-13 at 6 02 43 PM
Screen Shot 2021-09-13 at 6 04 23 PM
Screen Shot 2021-09-13 at 6 05 07 PM

### CODE REVIEW

File: frontend/public/components/namespace.jsx [hunk 1]
```diff
@@ -1111,7 +1111,7 @@ export const NamespaceDetails = ({ obj: ns, customData }) => {
 
 const RolesPage = ({ obj: { metadata } }) => (
   <RoleBindingsPage
-    createPath={`/k8s/ns/${metadata.name}/rolebindings/~new?rolekind=Role`}
+    createPath={`/k8s/ns/${metadata.name}/rolebindings/~new?rolekind=ClusterRole`}
```
Comment (spadgett) [inline frontend/public/components/namespace.jsx:1114]: Hey, @cyril-ui-developer. I'm unclear why this change was needed.

Comment (cyril-ui-developer) [reply frontend/public/components/namespace.jsx:1114]: > Hey, @cyril-ui-developer. I'm unclear why this change was needed.

@spadgett The objects in `sortedList` which the role dropdown make use of are concatenated with `xxxx-ClusterRole`. So if `kindLabel` is  set to `Role` there is not match. That was my thought for changed.
https://github.com/openshift/console/blob/master/frontend/public/components/utils/list-dropdown.jsx#L110
Also, the role dropdown items are from `/clusterroles` endpoint. 

However, I investigated further, and found out that switching OR operator operands fix the issue. 
https://github.com/openshift/console/blob/master/frontend/public/components/RBAC/bindings.jsx#L645
` selectedKeyKind={roleRef.kind || _.get(fixed, 'roleRef.kind')}`
Because the `roleRef.kind = "ClusterRole"` and `_.get(fixed, 'roleRef.kind') = "Role"` if the context is from Project.  


I think the `roleRef.kind` and  `fixed.roleRef.kind` should be the same value or one is present and the other undefined. 
If the context is from User Mgt,  the `fixed.roleRef.kind` is `null`

**Comment by cyril-ui-developer** [comment]
@spadgett Updated! Thanks.

File: frontend/public/components/namespace.jsx (hunk 1 above)
Comment (spadgett) [reply frontend/public/components/namespace.jsx]: Hey, @cyril-ui-developer. Sorry for not responding sooner.

I think the mistake is that we're passing a `roleKind` at all? If I'm reading the code right, `roleKind` should only be used with a specific role name. We should be able to handle both cluster roles and namespaced roles here, and we should make sure both work.

Comment (cyril-ui-developer) [reply frontend/public/components/namespace.jsx]: > Hey, @cyril-ui-developer. Sorry for not responding sooner.
> 
> I think the mistake is that we're passing a `roleKind` at all? If I'm reading the code right, `roleKind` should only be used with a specific role name. We should be able to handle both cluster roles and namespaced roles here, and we should make sure both work.

@spadgett No worries. 

If I got you right , the `roleKind` should be pass if the context is from Roles page only. 

So we need to remove the `roleKind` in the  namespace url query to fix the issue.

Comment (spadgett) [reply frontend/public/components/namespace.jsx]: I believe that's right.

Comment (cyril-ui-developer) [reply frontend/public/components/namespace.jsx]: > I believe that's right.

@spadgett Updated. PTAL.

**Review by spadgett — APPROVED** [review]
/lgtm
